bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ChannelItemProvider.java

```diff
@@ -174,6 +165,18 @@ protected synchronized void modified(Map<String, Object> properties) {
         }
     }
 
+    private void delayedInitialize(ScheduledExecutorService executor) {
+        // we wait until no further new links or items are announced in order to avoid creation of
+        // items which then must be removed again immediately.
+        final long diff = System.nanoTime() - lastUpdate - TimeUnit.SECONDS.toNanos(2);
+        if (diff < 0) {
```

Comment:
Ok, thanks - I somehow had missed that you schedule `delayedInitialize` itself again.